api/queries_pr_review.go

```diff
@@ -0,0 +1,117 @@
+package api
+
+import (
+	"context"
+	"time"
+
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/shurcooL/githubv4"
+)
+
+type PullRequestReviewState int
+
+const (
+	ReviewApprove PullRequestReviewState = iota
+	ReviewRequestChanges
+	ReviewComment
+)
+
+type PullRequestReviewInput struct {
+	Body  string
+	State PullRequestReviewState
+}
+
+type PullRequestReviews struct {
+	Nodes      []PullRequestReview
+	PageInfo   PageInfo
+	TotalCount int
+}
+
+type PullRequestReview struct {
+	Author              Author
+	AuthorAssociation   string
+	Body                string
+	CreatedAt           time.Time
+	IncludesCreatedEdit bool
+	ReactionGroups      ReactionGroups
+	State               string
+	URL                 string
+}
+
+func AddReview(client *Client, repo ghrepo.Interface, pr *PullRequest, input *PullRequestReviewInput) error {
+	var mutation struct {
+		AddPullRequestReview struct {
+			ClientMutationID string
+		} `graphql:"addPullRequestReview(input:$input)"`
+	}
+
+	state := githubv4.PullRequestReviewEventComment
+	switch input.State {
+	case ReviewApprove:
+		state = githubv4.PullRequestReviewEventApprove
+	case ReviewRequestChanges:
+		state = githubv4.PullRequestReviewEventRequestChanges
+	}
+
+	body := githubv4.String(input.Body)
+	variables := map[string]interface{}{
+		"input": githubv4.AddPullRequestReviewInput{
+			PullRequestID: pr.ID,
+			Event:         &state,
+			Body:          &body,
+		},
+	}
+
+	gql := graphQLClient(client.http, repo.RepoHost())
+	return gql.MutateNamed(context.Background(), "PullRequestReviewAdd", &mutation, variables)
+}
+
+func pullRequestReviewsFragment() string {
+	return `reviews(last: 100) {
```

Comment:
This caps the number of displayed reviews at a maximum of 100. That might be fine, but we could also consider explicitly paginating the reviews collection to ensure that we always load all of them like we do with comments. If we don't, I'd like to see this cap of 100 be represented more explicitly in code.

Reply:
I went back and forth on if I should replicate retrieving of reviews the same way we do for comments. I chose this way for two reasons. 

First, I felt that it would be extremely rare for a PR to have over 100 reviews, where as it seems more likely that a PR/issue would have over 100 comments. 

Second, due to the fact that we display a list of reviewers by default without specifying the `--comments` flag we need to retrieve the reviews for every invocation of `pr view` and I wanted to save us from making an extra api call(s) each time. This may be a premature optimization.

Can you expand a bit about representing the cap more explicitly in code? Would creating a constant called `PullRequestReviewLimit` be explicit enough?

Reply:
> First, I felt that it would be extremely rare for a PR to have over 100 reviews, where as it seems more likely that a PR/issue would have over 100 comments.

I wouldn't say that it's that rare! For example, this PR already has 13 reviews, most of which were generated by only 2 people in the last 6 hours. (Note that each of our one-off replies in thread generates a separate review.) Some PRs in larger projects undergo weeks of reviews by a larger number of people.

> Would creating a constant called `PullRequestReviewLimit` be explicit enough?

Yes that would be great!

Reply:
I didn't realize every one-off response would create a new review. Due to that, I agree that a PR with >100 reviews is more likely than I originally thought. I changed the review retrieval code to match the comment retrieval code which properly handles >100 reviews.